README.md

```diff
@@ -1,66 +1,124 @@
-
-Supported types and values for the deserialization:
- - `()` from `undefined` and `null`.
- - `Option` from any value will map `undefined` or `null` to `None` and any other value to `Some(...)`.
- - `bool` from a JavaScript boolean (`false` and `true`).
- - Rust integer (`u8`/`i8`/.../`u128`/`i128`) from a safe JavaScript integer (as matched by [`Number.isSafeInteger`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isSafeInteger)).
- - Rust floating number (`f32`/`f64`) from any JavaScript number.
- - `char` from a JavaScript string containing a single codepoint.
- - `String` from any JavaScript string.
- - Rust map (`HashMap`, `BTreeMap`, ...) from any JavaScript iterable producing `[key, value]` pairs (including but not limited to ES2015 `Map`).
-   > One exception being [internally tagged](https://serde.rs/enum-representations.html#internally-tagged) and [untagged](https://serde.rs/enum-representations.html#untagged) enums. These representations currently do not support deserializing map-like iterables. They only support deserialization from `Object` due to their special treatment in `serde`. 
-   >
-   > This restriction may be lifted at some point in the future if a `serde(with = ...)` attribute can define the expected Javascript representation of the variant, or if serde-rs/serde#1183 gets resolved.
- - `HashMap<String, _>` from any plain JavaScript object (`{ key1: value1, ... }`).
- - Rust sequence (tuple, `Vec`, `HashSet`, ...) from any JavaScript iterable (including but not limited to `Array`, ES2015 `Set`, etc.).
- - Rust byte buffer (see [`serde_bytes`](https://github.com/serde-rs/bytes)) from JavaScript `ArrayBuffer` or `Uint8Array`.
- - Typed Rust structure from any plain JavaScript object (`{ key1: value1, ... }`).
- - Rust enum from either a string (`"Variant"`) or a plain object. Specific representation is [controlled](https://serde.rs/enum-representations.html) by `#[serde(...)]` attributes and should be compatible with `serde-json`.
-
-Serialization is compatible with the deserialization, but it's limited to a single representation, so it chooses:
- - `undefined` for `()` or `None`.
- - ES2015 `Map` for Rust maps (can be configured to use plain objects via `serialize_maps_as_objects(true)`).
- - `Array` for any Rust sequences.
- - `Uint8Array` for byte buffers.
- - Plain JavaScript object for typed Rust structures.
+Likewise, the `from_value` function can be used for deserialization.
+
+## Supported Types
+
+Note that, even though it might often be the case, this library doesn't attempt
+to be strictly compatible with either [`serde_json`][serde_json] or,
+correspondingly, `JsValue::from_serde` / `JsValue::into_serde`, instead
+prioritising better compatibility with common JavaScript idioms and
+representations.
+
+[serde_json]: https://docs.serde.rs/serde_json/
+
+### Deserialization
+
+| Javascript Type                            | Rust Type                          |
```

Comment:
I like the table idea overall. I wonder though, if we go there, perhaps it's better to merge both tables into one, with one column for "idiomatic type" (both used for serialization and supported for deserialization) and another column for "other supported sources" (extra types supported for deserialization only).

Reply:
Sure I'm in favour of that.

Reply:
Are you planning to come back to this?

Reply:
After looking at it again, fusing the charts might be harder than we initially thought. Each serialization direction has information unique to it, as well as caveats, so maybe it's clearer to keep them separate?